src/linter/html/HtmlReporter.ts

```diff
@@ -0,0 +1,84 @@
+import type {BaseReporter, ReporterMessage, ReporterCoverageInfo} from "../../detectors/BaseReporter.js";
+import type {LintMessage} from "../../detectors/AbstractDetector.js";
+import {Tag as SaxTag} from "sax-wasm";
+import {LintMessageSeverity, CoverageInfo} from "../../detectors/AbstractDetector.js";
+import {resolveLinks} from "../../formatter/lib/resolveLinks.js";
+
+export default class HtmlReporter implements BaseReporter {
+	#filePath: string;
+	#messages: LintMessage[] = [];
+	#coverageInfo: CoverageInfo[] = [];
+
+	constructor(filePath: string) {
+		this.#filePath = filePath;
+	}
+
+	addMessage({node, message, messageDetails, severity, ruleId, fatal = undefined}: ReporterMessage) {
+		if (fatal && severity !== LintMessageSeverity.Error) {
+			throw new Error(`Reports flagged as "fatal" must be of severity "Error"`);
+		}
+
+		let line = 1, column = 1;
```

Comment:
```suggestion
		let line = 0, column = 0;
```

Probably? Since we always do `+ 1` later

Reply:
Ah, thanks!